### ISSUE TITLE
Check date handling everywhere dates are displayed before BST kicks in

### ISSUE THREAD

**Issue by ismail-s** [ismail-s/JTime#253]
# Initial description
I'm looking at the dates in the Android app and on the website and just getting really confused. I think in some places I am handling salaah times incorrectly. I think this is in the website, both in saving and in displaying salaah times. But magrib times seem to be affected too in the Android app, so I need to check that the times being returned by GeoNames are correct. If they are not, then we have a problem there too.

_This bug has the highest priority._ Once it is fixed (and carefully checked), then a release is to be cut immediately.

**Deadline for resolution**: 24/3/2017 (ie before BST starts). Preferably a lot sooner though. Note that this bug is already present in the live Android app/website.

# Notes as this issue is worked through
## Non-magrib times/issues pertaining to all salaah times
- I need to check code that adds and displays salaah times, in all components.
  - User-inputted times should be checked and sent to the rest api with no timezone modifications.
  - The rest api should also make no timezone modifications when serving times.
  - Times should be displayed as they are received from the rest api, with no timezone modifications
  - To put it simply, _all code should essentially ignore timezones_, except magrib times (where timezones are currently dealt with by external apis called by the rest api). The user-inputted times have already had timezone modifications applied by the end-user.

## Magrib times
- The geonames api is returning the wrong time on 26 March 2017 for the UK. The time it is returning is in GMT, not BST. But this is for one day. So in the future, we should switch to another service, eg https://askgeo.com/ or use https://github.com/mourner/suncalc with some timezone determining method (see #254).
- The code in the REST API was parsing the sunset times according to server local time. This has been fixed and manually tested.
